inputremapper/gui/data_manager.py

```diff
@@ -0,0 +1,446 @@
+        logger.info('Removing "%s"', preset_path)
+        os.remove(preset_path)
+        self._active_mapping = None
+        self._active_preset = None
+        self._send_group()
+
+    def update_mapping(self, **kwargs):
+        """update the active mapping with the given keywords and values.
+
+        Will send "mapping" message to the DataBus. In case of a new event_combination
+        this will first send a "combination_update" message
+        """
+        if not self._active_mapping:
+            raise DataManagementError("Cannot modify Mapping: mapping is not set")
+
+        combination = self.active_mapping.event_combination
+        for key, value in kwargs.items():
+            setattr(self._active_mapping, key, value)
+
+        if (
+            "event_combination" in kwargs
+            and combination != self.active_mapping.event_combination
+        ):
+            self.data_bus.send(
+                CombinationUpdate(combination, self._active_mapping.event_combination)
+            )
+        self._send_mapping()
+
+    def create_mapping(self):
+        """create empty mapping in the active preset.
+        Will send "preset" message to the DataBus
+        """
+        if not self._active_preset:
+            raise DataManagementError("cannot create mapping: preset is not set")
+        self._active_preset.add(UIMapping())
+        self._send_preset()
+
+    def delete_mapping(self):
+        """delete the active mapping
+        Will send "preset" message to the DataBus
+        """
+        if not self._active_mapping:
+            raise DataManagementError(
+                "cannot delete active mapping: active mapping is not set"
+            )
+
+        self._active_preset.remove(self._active_mapping.event_combination)
+        self._active_mapping = None
+        self._send_preset()
+
+    def send_uinputs(self):
+        """send the "uinputs" message on the DataBus"""
+        self.data_bus.send(
+            UInputsData(
+                {
+                    name: uinput.capabilities()
+                    for name, uinput in self._uinputs.devices.items()
+                }
+            )
+        )
+
+    def send_groups(self):
+        """send the "groups" message on the DataBus"""
+        self._reader.emit_groups_changed()
+
+    def save(self):
+        """save the active preset"""
+        if self._active_preset:
+            self._active_preset.save()
+
+    def refresh_groups(self):
+        """refresh the groups (plugged devices)
+        Should send "groups" message to DataBus this will not happen immediately
+        because the system might take a bit until the groups are available
+        """
+        self._reader.refresh_groups()
+
+    def start_combination_recording(self):
+        """recorde user input
+        Will send "combination_recorded" messages as new input arrives.
+        Will eventually send a "recording_finished" message.
+        """
+        self._reader.start_recorder()
```

Comment:
is it a reader or a recorder?

Reply:
The class is a reader which reads all messages from the helper.
It has a recorder which can be started to record User input.

we can also call the whole thing a recorder or reader. whatever you prefer.

Reply:
what about `HelperClient`?